src/react/ReactElementSet.js

```diff
@@ -94,23 +94,26 @@ export default class ReactElementSet {
   }
 
   // for objects: [key/symbol] -> [key/symbol]... as nodes
-  _getObjectValue(object: ObjectValue): ObjectValue {
+  _getObjectValue(object: ObjectValue, visitedValues: Set<Value>): ObjectValue {
+    if (visitedValues.has(object)) return object;
```

Comment:
In some cases, not the incoming `object`, but some other value is returned. Why is it the right thing to do for previously visited values to return the incoming value?